### PR TITLE
OJ-28807-at-py-test-via-pdm

### PR DESCRIPTION
**Pull request by gavinpitt-jf**
Add in the `pytest` library to PDM. Also clean up a few problems detected when testing

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [tool.black]
 line-length = 100
 skip-string-normalization = true
-target_version = ['py310']
+target_version = ['py37', 'py38']
```
Comment (gavinpitt-jf) [inline pyproject.toml:4]: @ametzger as noted in slack, I'm rolling this back. I think we need to update `black` before we can use it with py310?

File: tests/test_jira_download.py [hunk 2]
```diff
@@ -90,18 +100,24 @@ def one_bad_issue_callback(self, request, context):
     @classmethod
     def setUpClass(cls):
         cls.jira_connection = get_connection()
-        with open("test_data/jira/test_issues_response.json", "r") as issues_file:
+        with open("tests/test_data/jira/test_issues_response.json", "r") as issues_file:
```
Comment (gavinpitt-jf) [inline tests/test_jira_download.py:103]: @thingscouldbeworse - most of the files changes here are formatting, but this was an intentional change. For some reason this base file path was missing in this test. It was causing errors for me when run on the command line

File: pyproject.toml [hunk 3]
```diff
@@ -67,6 +67,7 @@ dependencies = [
     "black~=19.10b0",
     "click~=8.0.4",
     "requests>=2.31.0",
+    "pytest>=7.4.0",
```
Comment (acicchetti-jf) [inline pyproject.toml:70]: Should this be a dev dependency?

Comment (gavinpitt-jf) [reply pyproject.toml:70]: Yes

**Review by thingscouldbeworse — APPROVED** [review]

**Review by acicchetti-jf — APPROVED** [review]

**Pull request merged**